src/libs/webview/request_dispatcher.cpp

```diff
@@ -326,13 +326,14 @@ WebRequestDispatcher::queue_basic_auth_fail(struct MHD_Connection *connection, W
 	sreply.pack();
 	struct MHD_Response *response = prepare_static_response(&sreply);
 
-	int rv = MHD_queue_basic_auth_fail_response(connection, realm_, response);
+	MHD_RESULT rv =
+	  static_cast<MHD_RESULT>(MHD_queue_basic_auth_fail_response(connection, realm_, response));
```

Comment:
OK right, a simple cast removes the warning, but it shouldn't be there. This makes the code more unsafe. Of course in this case it's unlikely that upstream will change the return type yet again, but IF they do, this actually becomes dangerous because the compiler will silently ignore various type mismatches and allow overruns.

Reply:
For context, this is the problem: https://bugs.gnunet.org/view.php?id=6441 
Maybe they'll fix it quickly if we make some noise there^^

Reply:
Oh and by "change yet again" of course I don't mean changing the `int` to `MHD_Result`, which they should do ASAP. What I mean is if at a later point they decide that `MHD_Result` doesn't cut it and they need something else.